server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java

```diff
@@ -130,18 +132,27 @@ public class SystemIndexDescriptor implements IndexPatternMatcher, Comparable<Sy
 
     /**
      * For internally-managed indices, specifies a key name under <code>_meta</code> in the index mappings
-     * that contains the index's mappings' version.
+     * that contains the index's mappings' {@link Version}. We need to read and write this field for
+     * backwards compatibility.
      */
-    private final String versionMetaKey;
+    private final String mappingsNodeVersionMetaKey;
+
+    /** The version meta key for the integer system index mapping version */
+    public static final String VERSION_META_KEY = "system_index_mappings_version";
```

Comment:
That's a good suggestion. I used a long key name in order to avoid clashing with any of the different version meta keys that are being used out there. I think I might go with `managed_index_mappings_version`, because the system index code uses the "internal/external" terms to specify whether an external client can interact with them. Fleet indices are external-managed, while the tasks index is internal-managed. What do you think?

Reply:
Yes, `managed_index_mappings_version` sounds good to me. Thanks!